### PR TITLE
spin github action

### PR DESCRIPTION
**Pull request by rajatjindal**

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,22 @@
+
+The MIT License (MIT)
```
Comment (radu-matei) [inline LICENSE]: We default to Apache 2 as the license for new projects.
Any reason to go for MIT here?

File: action.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+name: 'setup-spin'
+description: 'downloads and add spin to your PATH'
+author: 'Rajat Jindal'
```
Comment (radu-matei) [inline action.yml]: `author: Fermyon Engineering`

File: action.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    description: 'path to spin.toml file'
+  cloud_base_url:
+    required: false
+    description: 'cloud base url to use. defaults to https://cloud.fermyon.com'
+  oci_app_reference:
+    required: false
+    description: 'oci reference for pushing spin app to oci registry. e.g. ghcr.io/rajatjindal/cloud-start:v0.0.1'
+  deploy:
+    required: false
+    description: 'deploys the app to Fermyon cloud'
```
Comment (radu-matei) [inline action.yml]: `Fermyon Cloud`

File: action.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    type: boolean
+  app_config_file:
+    required: false
+    description: 'path to spin.toml file'
+  cloud_base_url:
+    required: false
+    description: 'cloud base url to use. defaults to https://cloud.fermyon.com'
+  oci_app_reference:
+    required: false
+    description: 'oci reference for pushing spin app to oci registry. e.g. ghcr.io/rajatjindal/cloud-start:v0.0.1'
```
Comment (radu-matei) [inline action.yml]: `*Registry* reference`
I would also point to the Spin docs for more info here?

File: src/downloader.ts [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+
+export function getConfig(
+  nameInput: string,
+  urlInput: string,
+  pathInArchive: string
+): Downloader {
+  return new Downloader(nameInput, urlInput, pathInArchive)
+}
+
+export class Downloader {
```
Comment (radu-matei) [inline src/downloader.ts:31]: Is there any reason to re-implement and maintain all of this code here?
I think it is already part of actions we use and we could import.
(not blocking this PR, merely a suggestion for the future)

File: LICENSE (hunk 1 above)
Comment (rajatjindal) [reply LICENSE]: changed to Apache. I had taken some code from a MIT licensed repo (`configurator`), so licensed it similarly. Did some reading on that  and we can still do Apache, so changed it Apache now.

File: action.yml (hunk 2 above)
Comment (rajatjindal) [reply action.yml]: done.

File: action.yml (hunk 3 above)
Comment (rajatjindal) [reply action.yml]: done

File: action.yml (hunk 4 above)
Comment (rajatjindal) [reply action.yml]: done

File: src/downloader.ts (hunk 5 above)
Comment (rajatjindal) [reply src/downloader.ts:31]: I think we need to publish that as `npm package` to be able to use it here. (we can import using git reference as well, but it is not recommended). I will talk to @karthik2804 about it.

Comment (rajatjindal) [reply src/downloader.ts:31]: also for github action, we don't need to support all platform/arch. so we can probably simplify that code into a function and just include that here. but open to suggestions here.

**Review by radu-matei — APPROVED** [review]
This is in a great spot, we can improve things along the way (like reusing parts of the download logic from a shared package), but this LGTM!

**Pull request merged**